Clang 16: incompatible initializer of type (container_common_test)

Compilation of *container_common_test.cpp* fails on Clang 16:

```
 /__w/FunctionalPlus/FunctionalPlus/test/container_common_test.cpp:664:16: error: variable 'f' with type 'const auto' has incompatible initializer of type '<overloaded function type>'
    const auto f = floor<double>;
               ^   ~~~~~~~~~~~~~
```

Full build log is available on Clang 16 PR (#274).

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -38,6 +38,7 @@ jobs:
     strategy:
       matrix:
         build_config:
+          - { version: 16 }
           - { version: 15 }
           - { version: 14 }
           - { version: 13 }

--- a/test/container_common_test.cpp
+++ b/test/container_common_test.cpp
@@ -661,7 +661,7 @@ TEST_CASE("container_common_test - coucount_occurrences_bynt_occurrences_on")
     typedef std::map<int, std::size_t> IntSizeTMap;
     IntSizeTMap OccurrencesResult = {{1, 1}, {2, 3}, {3, 1}};
     std::vector<double> double_values = {1.1, 2.3, 2.7, 3.6, 2.4};
-    const auto f = floor<double>;
+    const auto f = fplus::floor<double>;
     REQUIRE_EQ(count_occurrences_by(f, double_values), OccurrencesResult);
 }
 

